nixops/backends/__init__.py

```diff
@@ -212,7 +212,10 @@ def send_keys(self):
         for k, opts in self.get_keys().items():
             self.log("uploading key ‘{0}’...".format(k))
             tmp = self.depl.tempdir + "/key-" + self.name
-            f = open(tmp, "w+"); f.write(opts['text']); f.close()
+
+            if   'text'     in opts: f = open(tmp, "w+"); f.write(opts['text']); f.close()
+            elif 'filename' in opts: self._logged_exec(["cp", opts['filename'], tmp])
```

Comment:
Or use Python's [shutils](https://docs.python.org/3/library/shutil.html) or something like that. Absolute paths are still good :)